packages/utils/src/utils/arrays.utils.ts

```diff
@@ -57,5 +57,12 @@ export const hexStringToUint8Array = (hexString: string): Uint8Array => {
   return Uint8Array.from(matches.map((byte) => parseInt(byte, 16)));
 };
 
-export const uint8ArrayToHexString = (bytes: Uint8Array) =>
-  bytes.reduce((str, byte) => str + byte.toString(16).padStart(2, "0"), "");
+export const uint8ArrayToHexString = (bytes: Uint8Array | number[]) => {
+  if (!(bytes instanceof Uint8Array)) {
```

Comment:
strictly out of curiosity, would actually `Uint8Array.from(aUint8Array)` work out or does this throw an error?

Reply:
That works fine. But I didn't want to allocate a new Uint8Array if not necessary. But also not super important.

Reply:
Good to know! Thanks for the feedback.